Not `no_std` friendly

While this compiles to wasm, with or without std. if you have a custom panic handler, it will fail with the following error:

```rs
error[E0152]: found duplicate lang item `panic_impl`
 --> wasm/nostd/src/lib.rs:9:1
  |
9 | fn panic(_panic: &PanicInfo<'_>) -> ! {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: the lang item is first defined in crate `std` (which `subtle` depends on)
  = note: first definition in `std` loaded from /nix/store/przz0bb6zljm8sbrdavvk6d28pwgj5fx-rust-std-1.75.0-wasm32-unknown-unknown/lib/rustlib/wasm32-unknown-unknown/lib/libstd-0a1cbf5fdf3ff00d.rlib
  = note: second definition in the local crate (`nostd_example`)

For more information about this error, try `rustc --explain E0152`.
```

*Minimum Example*

```toml
[package]
name = "nostd-example"
version = "0.1.0"
edition = "2021"
publish = false

﻿﻿[dependencies]
stark-curve = { version = "0.1", default-features = false }

```

```rs
#![no_std]

use ::core::panic::PanicInfo;

pub use stark_curve::*;

#[panic_handler]
fn panic(_panic: &PanicInfo<'_>) -> ! {
    loop {}
}

```

**The root of the issue**
The issue is from the `subtle` crate, we need to disable the default feature for it.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,11 +11,11 @@ keywords = ["elliptic-curves", "stark-curve"]
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [dependencies]
-ff = { version = "0.13", features = ["derive"] }
+ff = { version = "0.13", default-features = false, features = ["derive"] }
 hex-literal = "0.3"
 primeorder = "0.13"
-subtle = "2"
-zeroize = "1.5"
+subtle = { version = "2", default-features = false }
+zeroize = { version = "1.5", default-features = false }
 
 [dev-dependencies]
 rand = "0.8"

